src/server/game/Entities/Player/Player.cpp

```diff
@@ -14962,6 +14967,10 @@ void Player::CompleteQuest(uint32 quest_id)
         if (Quest const* qInfo = sObjectMgr->GetQuestTemplate(quest_id))
             if (qInfo->HasFlag(QUEST_FLAGS_TRACKING))
                 RewardQuest(qInfo, 0, this, false);
+
+        // area auras may change on quest completion
+        UpdateZoneDependentAuras(GetZoneId());
+        UpdateAreaDependentAuras(GetAreaId());
```

Comment:
I've noted in my to do list to check what's going on there but forgot to check it before pushing. It should do the same thing, you are right